src/App.tsx

```diff
@@ -16,35 +16,92 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [isSortedAlphabetically, setIsSortedAlphabetically] = useState<boolean>(false);
+  const [isSortedByLength, setIsSortedByLength] = useState<boolean>(false);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const sortAlphabetically = (goods: string[]): string[] => ([...goods].sort((a, b) => a.localeCompare(b)));
+
+  const sortByLength = (goods: string[]): string[] => ([...goods].sort((a, b) => a.length - b.length));
+
+  const reverseOrder = (goods: string[]): string[] => ([...goods].reverse());
+
+  const reset = (): void => {
+    setIsSortedAlphabetically(false);
+    setIsSortedByLength(false);
+    setIsReversed(false);
+  };
+
+  const isResetVisible: boolean = isSortedAlphabetically || isSortedByLength || isReversed;
+
+  const getVisibleGoods = (): string[] => {
```

Comment:
You can move all these functions outside the component `getVisibleGoods`, `sortAlphabetically`, `sortByLength`, `reverseOrder`